concourse/scripts/pipelines-bosh-cloudfoundry.sh

```diff
@@ -116,38 +116,47 @@ generate_manifest_file() {
     < "${SCRIPT_DIR}/../pipelines/${pipeline_name}.yml"
 }
 
+upload_pipeline() {
+  bash "${SCRIPT_DIR}/deploy-pipeline.sh" \
+        "${env}" "${pipeline_name}" \
+        <(generate_manifest_file) \
+        <(generate_vars_file)
+}
+
+remove_pipeline() {
+  yes y | ${FLY_CMD} -t "${FLY_TARGET}" destroy-pipeline --pipeline "${pipeline_name}" || true
+}
+
 update_pipeline() {
   pipeline_name=$1
 
   case $pipeline_name in
-    create-bosh-cloudfoundry|failure-testing)
-      bash "${SCRIPT_DIR}/deploy-pipeline.sh" \
-        "${env}" "${pipeline_name}" \
-        <(generate_manifest_file) \
-        <(generate_vars_file)
+    create-bosh-cloudfoundry)
+      upload_pipeline
+    ;;
+    failure-testing)
+      if [ -n "${ENABLE_FAILURE_TESTING:-}" ] && [ "${ENABLE_FAILURE_TESTING:-}" = "true" ]; then
```

Comment:
In this line, the first test is redundant. This could simply be `if [ "${ENABLE_FAILURE_TESTING:-}" = "true" ];`

Having said that, see my comment below about the interpretation of these flags.